pageserver/src/walreceiver.rs

```diff
@@ -204,7 +204,13 @@ fn walreceiver_main(
     let end_of_wal = Lsn::from(u64::from(identify.xlogpos));
     let mut caught_up = false;
 
-    let timeline = tenant_mgr::get_timeline_for_tenant(tenantid, timelineid)?;
+    let timeline =
+        tenant_mgr::get_timeline_for_tenant(tenantid, timelineid).with_context(|| {
```

Comment:
Maybe makes sense to have `get_local_timeline_for_tenant` because there are now a few such places

Reply:
Yeah, I've decided to not to go with that since I'm not fond of panicking, and whatever else I return (`Option`?), I have to process somehow, so same few lines everywhere around.

Reply:
I think there is no panicking required, IMO It makes sense for it to be a Result, which carries both, absence of a timeline and timeline being remote instead of local. And no more code lines required, just `?` and maybe `.context`

This is a nitpick so feel free to leave as is